float32 relativeX
float32 relativeY
